docs/source/integrations/mlflow.rst

```diff
@@ -1,3 +1,342 @@
+frameworks:
+
+.. code-block:: bash
+
+    bentoml models list mlflow_pytorch_mnist
+
+
+The second argument ``model_uri`` takes a URI to the MLflow model. It can be a local
+path, a ``'runs:/'`` URI, or a remote storage URI (e.g., an ``'s3://'`` URI). Here are
+some example ``model_uri`` values commonly used in MLflow:
+
+.. code-block::
+
+    /Users/me/path/to/local/model
+    ../relative/path/to/local/model
+    s3://my_bucket/path/to/model
+    runs:/<mlflow_run_id>/run-relative/path/to/model
+    models:/<model_name>/<model_version>
+    models:/<model_name>/<stage>
+
+
+Running Imported Model
+----------------------
+
+MLflow models imported to BentoML can be loaded back for running inference in a various
+of ways.
+
+Loading original model flavor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For evaluation and testing purpose, sometimes it's convenient to load the model in its
+native form
+
+.. code-block:: python
+
+    bento_model = bentoml.mlflow.get("mlflow_pytorch_mnist:latest")
+    loaded_pytorch_model = mlflow.pytorch.load_model(bento_model.path_of("mlflow_model"))
+    loaded_pytorch_model.to(device)
+    loaded_pytorch_model.eval()
+    with torch.no_grad():
+        input_tensor = torch.from_numpy(test_input_arr).to(device)
+        predictions = loaded_pytorch_model(input_tensor)
+
+
+Loading pyfunc flavor
+~~~~~~~~~~~~~~~~~~~~~
+
+By default, ``bentoml.mflow.load_model`` will load the imported MLflow model using the
+`python_function flavor <https://www.mlflow.org/docs/latest/python_api/mlflow.pyfunc.html>`_
+for best compatibility across all ML frameworks supported by MLflow.
+
+.. code-block:: python
+
+    pyfunc_model: mlflow.pyfunc.PyFuncModel = bentoml.mlflow.load_model("mlflow_pytorch_mnist:latest")
+    predictions = pyfunc_model.predict(test_input_arr)
+
+
+Using Model Runner
+~~~~~~~~~~~~~~~~~~
+
+Imported MLflow models can be loaded as BentoML Runner for best performance in building
+prediction service with BentoML. To test out the runner API:
+
+.. code-block:: python
+
+    runner = bentoml.mlflow.get("mlflow_pytorch_mnist:latest").to_runner()
+    runner.init_local()
+    runner.predict.run(input_df)
+
+Learn more about BentoML Runner at :doc:`/concepts/runner`.
+
+Runner created from an MLflow model supports the following input types. Note that for
+some ML frameworks, only a subset of this list is supported.
+
+.. code-block:: python
+
+    MLflowRunnerInput = Union[pandas.DataFrame, np.ndarray, List[Any], Dict[str, Any]]
+    MLflowRunnerOutput = Union[pandas.DataFrame, pandas.Series, np.ndarray, list]
+
+.. note::
+
+    To use adaptive batching with a MLflow Runner, make sure to set
+    ``signatures={'predict': {'batchable': True}}`` when importing the model:
+
+    .. code-block:: python
+
+        bento_model = bentoml.mlflow.import_model(
+            'mlflow_pytorch_mnist',
+            model_uri,
+            signatures={'predict': {'batchable': True}}
+        )
+
+
+Limitations
+~~~~~~~~~~~
```

Comment:
Instead of limitations, let's phrase it as opportunity for performance optimization.